packages/@react-spectrum/tabs/test/Tabs.test.js

```diff
@@ -249,4 +261,71 @@ describe('Tabs', function () {
     userEvent.click(tabs[1]);
     expect(onSelectionChange).not.toBeCalled();
   });
+
+  it('collapses when it can\'t render all the tabs horizontally', function () {
+    jest.spyOn(HTMLElement.prototype, 'getBoundingClientRect').mockImplementationOnce(function () {
+      if (this instanceof HTMLDivElement) {
+        return {
+          right: 500
+        };
+      }
+    }).mockImplementationOnce(function () {
+      if (this instanceof HTMLDivElement) {
+        return {
+          right: 700
+        };
+      }
+    });
+
+    let {getByRole} = renderComponent({
+      'aria-label': 'Test Tabs',
+      'aria-labelledby': 'external label',
+      onSelectionChange,
+      defaultSelectedKey: items[0].name
+    });
+
+    expect(() => getByRole('tablist')).toThrow();
+    let tabpanel = getByRole('tabpanel');
+    expect(tabpanel).toBeTruthy();
+    expect(tabpanel).toHaveTextContent(items[0].children);
+
+    let picker = getByRole('button');
+    let pickerLabel = within(picker).getByText('Tab 1');
+    expect(picker).toHaveAttribute('aria-label', 'Test Tabs');
+    expect(picker).toHaveAttribute('aria-labelledby', `external label ${picker.id} ${pickerLabel.id}`);
+
+    triggerPress(picker);
+    act(() => jest.runAllTimers());
+    let listbox = getByRole('listbox');
+    let option = within(listbox).getByText('Tab 3');
+    triggerPress(option);
+    act(() => jest.runAllTimers());
+
+    expect(onSelectionChange).toBeCalledTimes(1);
+    expect(onSelectionChange).toHaveBeenCalledWith('Tab 3');
+
+    tabpanel = getByRole('tabpanel');
+    expect(tabpanel).toHaveTextContent(items[2].children);
+  });
+
+  it('doesn\'t collapse when it can render all the tabs horizontally', function () {
+    jest.spyOn(HTMLElement.prototype, 'getBoundingClientRect').mockImplementationOnce(function () {
+      if (this instanceof HTMLDivElement) {
+        return {
+          right: 500
+        };
+      }
+    }).mockImplementationOnce(function () {
+      if (this instanceof HTMLDivElement) {
+        return {
+          right: 400
+        };
+      }
+    });
+
+    let {getByRole} = renderComponent();
+    let tablist = getByRole('tablist');
+    expect(tablist).toBeTruthy();
+    expect(() => getByRole('button')).toThrow();
+  });
```

Comment:
I know this test isn't needed, but because of the storybook stories you wrote I want there to be a test that has the tabs collapse because you added tabs. :)

Reply:
hmm I can try doing a rerender call with a new mock on the boundingRect call